spec/controllers/cart_controller_spec.rb

```diff
@@ -1,86 +1,108 @@
-        end
+      describe "encoding Infinity" do
+        let!(:v) { create(:variant, on_demand: true, count_on_hand: 0) }
 
-        context 'an order with line items' do
-          let(:product) { FactoryBot.create(:product, distributors: [ distributor ]) }
-          let(:order) { FactoryBot.create(:order, { distributor: distributor } ) }
-          let(:line_item) { FactoryBot.create(:line_item, { variant: product.master }) }
-
-          before(:each) do
-            order.line_items << line_item
-            order.save
-            cart.orders << order
-            cart.save!
-          end
-
-          it "retrieves a cart with a single order and line item" do
-            get :show, {id: cart, :format => :json }
-            json_response = JSON.parse(response.body)
-
-            json_response['orders'].size.should == 1
-            json_response['orders'].first['distributor'].should == order.distributor.name
-            json_response['orders'].first['line_items'].first["name"].should == product.name
-            json_response['orders'].first['line_items'].first["quantity"].should == line_item.quantity
-          end
+        it "encodes Infinity as a large, finite integer" do
+          expect(controller.stock_levels(order, [v.id])).to eq(
+            {v.id => {quantity: 2, max_quantity: 3, on_hand: 2147483647}}
+          )
         end
+      end
+    end
+
+    it "extracts variant ids from the cart service" do
+      variants_h = [{:variant_id=>"900", :quantity=>2, :max_quantity=>nil},
+       {:variant_id=>"940", :quantity=>3, :max_quantity=>3}]
 
-        context 'adding a variant' do
+      expect(controller.variant_ids_in(variants_h)).to eq([900, 940])
+    end
+  end
 
-          it 'should add variant to new order and return the order' do
-            product1.distributors << distributor
-            product1.save
-            variant = product1.variants.first
+  context "adding a group buy product to the cart" do
+    it "sets a variant attribute for the max quantity" do
+      distributor_product = create(:distributor_enterprise)
+      p = create(:product, :distributors => [distributor_product], :group_buy => true)
 
-            put :add_variant, { cart_id: cart, variant_id: variant.id, quantity: (variant.on_hand-1), distributor_id: distributor, order_cycle_id: nil, max_quantity: nil }
+      order = subject.current_order(true)
+      allow(order).to receive(:distributor) { distributor_product }
+      expect(order).to receive(:set_variant_attributes).with(p.master, {'max_quantity' => '3'})
+      allow(controller).to receive(:current_order).and_return(order)
 
-            cart.orders.size.should == 1
-            cart.orders.first.line_items.size.should == 1
-            cart.orders.first.line_items.first.product.should == product1
-          end
-        end
-      end
+      expect do
+        spree_post :populate, :variants => {p.master.id => 1}, :variant_attributes => {p.master.id => {:max_quantity => 3}}
```

Comment:
Lots of unnecessary hash rockets here, I guess they were copied form an old spec? It's odd that codeclimate/rubocop didn't mention it.

Reply:
yes, existing spec from orders_controller.
good point, I dont think rubocop is going through this file...
To be honest I dont know how to fix this! These rules a tricky. I was reading this:
https://stackoverflow.com/questions/10004158/is-hash-rocket-deprecated

Original:
spree_post :populate, :variants => {p.master.id => 1}, :variant_attributes => {p.master.id => {:max_quantity => 3}}
Is this the better solution?
spree_post :populate, variants: { p.master.id => 1 }, variant_attributes: { p.master.id => {max_quantity: 3 } }

Reply:
I guess it does get a bit messy sometimes, like the single and double quotes.

Reply:
ok, done.